### PR TITLE
Accept any string as zipkinServiceName parameter.

### PR DESCRIPTION
**Pull request by rgamez**
While `remoteServiceName` accepts any string, the jdbc url parameter which provides a way to override it only accepts characters from the word group `[a-zA-Z0-9_]` this change remove that assumption and capture the complete value of the parameter and it solves a problem that I'm experiencing where in my naming convention (kebab case) is not possible to have the correct value due the presence of dash (`-`), for example `my-database-override` results in `my` as the `remoteServiceName`.

### CODE REVIEW

File: instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 final class TracingJdbcEventListener extends SimpleJdbcEventListener {
 
   private static final Pattern URL_SERVICE_NAME_FINDER =
-    Pattern.compile("zipkinServiceName=(\\w*)");
+    Pattern.compile("zipkinServiceName=(.*?)(?:&|$)");
```
Comment (jorgheymans) [inline instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java:33]: Could you maybe add a small comment what the regex is aiming to match ? Not everyone can work it out from the top of their head.

**Review by jorgheymans — CHANGES_REQUESTED** [review]
Thanks for this ! Could you maybe add a dedicated test with a few boundary testcases for the regex so it becomes clear what will match and what not ?

File: instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java (hunk 1 above)
Comment (rgamez) [reply instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java:34]: Comment included.

**Comment by rgamez** [comment]
Thanks @jorgheymans, I replaced the tests with some parametrized ones and included a few examples of some likely naming conventions and a few edge cases.

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java [hunk 2]
```diff
@@ -121,6 +128,34 @@
     verify(span).remoteIpAndPort("1.2.3.4", 5555);
   }
 
+  @Test
+  @TestCaseName("remoteServiceName should be '{1}'")
+  @Parameters({
```
Comment (adriancole) [inline instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:133]: this dep hasn't been updated since 2017, so might haunt us later. you mind switching to just using a map and calling a method not marked test for each pair here?

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java [hunk 3]
```diff
@@ -121,6 +128,34 @@
...
+    "?zipkinServiceName=my_database&foo=bar, my_database",
+    "?zipkinServiceName=my-database&foo=bar, my-database",
+    "?zipkinServiceName=my-database-1&foo=bar, my-database-1",
+    "?zipkinServiceName=my.database&foo=bar, my.database",
+    "?zipkinServiceName=my.database&foo=bar, my.database",
+    "?zipkinServiceName=my-database:5432&foo=bar, my-database:5432",
+    "?zipkinServiceName=my-database@localhost&foo=bar, my-database@localhost",
+    "?zipkinServiceName=my-database, my-database",
+    "?zipkinServiceName=my-database-1&zipkinServiceName=my-database-2, my-database-1",
+    "?zipkinServiceName=, default",
```
Comment (adriancole) [inline instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:144]: empty should coerce to null, not "default" IMHO

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java [hunk 4]
```diff
@@ -121,6 +128,34 @@
...
+    "?zipkinServiceName=myDatabase&foo=bar, myDatabase",
+    "?zipkinServiceName=my_database&foo=bar, my_database",
+    "?zipkinServiceName=my-database&foo=bar, my-database",
+    "?zipkinServiceName=my-database-1&foo=bar, my-database-1",
+    "?zipkinServiceName=my.database&foo=bar, my.database",
+    "?zipkinServiceName=my.database&foo=bar, my.database",
+    "?zipkinServiceName=my-database:5432&foo=bar, my-database:5432",
+    "?zipkinServiceName=my-database@localhost&foo=bar, my-database@localhost",
+    "?zipkinServiceName=my-database, my-database",
+    "?zipkinServiceName=my-database-1&zipkinServiceName=my-database-2, my-database-1",
```
Comment (adriancole) [inline instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:143]: am I reading this carefully? the service name would have a comma in it? that seems odd.. I think we should pick last instead.

**Review by adriancole — COMMENTED** [review]
thanks, the tests are telling!

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java (hunk 3 above)
Comment (rgamez) [reply instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:144]: bear in mind that this parameter is an override for `remoteServiceName`, not providing it just results in the default (the database name) used, that's the pre-existing behaviour.

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java (hunk 4 above)
Comment (rgamez) [reply instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:143]: that's just the separator between the two parameters queryString and the expected name.

File: instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java (hunk 2 above)
Comment (rgamez) [reply instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:133]: unfortunately for junit4 there are no better alternatives, junit5 has support for this style of tests out of the box, let me see what I can do.

Comment (adriancole) [reply instrumentation/p6spy/src/test/java/brave/p6spy/TracingJdbcEventListenerTest.java:133]: Arrays.asList in a for loop is better than a dead dependency. can you do add this? I'm worried about best being enemy of good..

**Comment by adriancole** [comment]
ps we ended up forced into jupiter, so after #1267 we will have it if you want to use it here

**Comment by adriancole** [comment]
I'll buff out the test dep after merge. thanks for the help!

**Comment by rgamez** [comment]
Great, came to volunteer for the jupiter port but I see that you already did it, thanks.

**Comment by adriancole** [comment]
incidentally rolled back the blanket jupiter as it would imply brave-tests which is a packaged jar, ending up staining classpaths of modules that import it. However, feel free to add jupiter (using the pinned version) to the "test" scope of brave-test if you like and port away!

what I did actually was use the junit4 experimental extension for the time being.
